app/helpers/textual_mixins/textual_refresh_status.rb

```diff
@@ -1,20 +1,29 @@
 module TextualMixins::TextualRefreshStatus
   def textual_refresh_status
-    last_refresh_status = @record.last_refresh_status.titleize
-    if @record.last_refresh_date
-      last_refresh_date = time_ago_in_words(@record.last_refresh_date.in_time_zone(Time.zone)).titleize
+    refresh_status(@record)
+  end
+
+  def textual_refresh_date
+    refresh_date(@record)
+  end
+
+  def refresh_status(record)
+    last_refresh_status = record.last_refresh_status ? record.last_refresh_status.titleize : nil
+    if record.last_refresh_date 
+      last_refresh_date = !record.last_refresh_date ? _('Never'): time_ago_in_words(record.last_refresh_date.in_time_zone(Time.zone)).titleize
       last_refresh_status << _(" - %{last_refresh_date} Ago") % {:last_refresh_date => last_refresh_date}
     end
     {
-      :label => _("Last Refresh"),
+      :label => _("Last Refresh Status"),
       :value => [{:value => last_refresh_status},
-                 {:value => @record.last_refresh_error.try(:truncate, 120)}],
-      :title => @record.last_refresh_error
+                 {:value => record.last_refresh_error.try(:truncate, 120)}],
+      :title => record.last_refresh_error
     }
   end
 
-  def textual_refresh_date
-    last_refresh_date = @record.last_refresh_date ? @record.last_refresh_date.in_time_zone(Time.zone) : nil
+  def refresh_date(record)
+    p "record.last_refresh_date-------> #{record.name}"
+    last_refresh_date = record.last_refresh_date ? record.last_refresh_date.in_time_zone(Time.zone) : nil
```

Comment:
Can also be written as:

```suggestion
   last_refresh_date = record.last_refresh_date && record.last_refresh_date.in_time_zone(Time.zone)
```

Reply:
@Fryguy can't it just be `last_refresh_date = record.last_refresh_date&.in_time_zone(Time.zone)` ?